test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -66,6 +66,67 @@ public HubConnectionTests(ServerFixture<Startup> serverFixture, ITestOutputHelpe
             }
         }
 
+
+        [Theory]
+        [MemberData(nameof(HubProtocolsAndTransportsAndHubPaths))]
+        public async Task CheckFixedMessage_HubConnectionFactory(IHubProtocol protocol, TransportType transportType, string path)
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                var connection = HubConnectionFactory.Create(new Uri(_serverFixture.BaseUrl + path), transportType, protocol, loggerFactory);
+                try
+                {
+                    await connection.StartAsync().OrTimeout();
+
+                    var result = await connection.InvokeAsync<string>("HelloWorld").OrTimeout();
+
+                    Assert.Equal("Hello World!", result);
+                }
+                catch (Exception ex)
+                {
+                    loggerFactory.CreateLogger<HubConnectionTests>().LogError(ex, "Exception from test");
+                    throw;
+                }
+                finally
+                {
+                    await connection.DisposeAsync().OrTimeout();
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(HubProtocolsAndTransportsAndHubPaths))]
+        public async Task CheckFixedMessage_HubConnectionBuilder(IHubProtocol protocol, TransportType transportType, string path)
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                var connection =
+                    new HubConnectionBuilder(new Uri(_serverFixture.BaseUrl + path))
+                        .WithTransportType(transportType)
+                        .WithHubProtocol(protocol)
+                        .WithLogger(loggerFactory)
+                        .Build();
```

Comment:
Shows the usage.

Reply:
I prefer the builder. It gels with everything else we've build. If you take it further there's a more general purpose client builder API here. Lets go with this.

Reply:
Agreed. Of the two I prefer the Builder.